### PR TITLE
POLIO-1011 fix group dropdown in campaigns page

### PR DESCRIPTION
**Pull request by olethanh**
Now we don't need the group permission to have the Group dropdown.
And it works in the anonymous calendar

Related JIRA tickets : POLIO-2011

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [x] My migrations file are included
- [x] Are there enough tests
- [x] Documentation has been included (for new feature)

## Changes

Added a new API for group that we can use later in dropdowns: `/api/groups/dropdown/`

## How to test

- Test the polio interface with an user without the group permission
- Test the embeddedCalendar in an private page.

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/orgUnits/hooks/requests/useGetGroups.ts [hunk 1]
```diff
@@ -55,3 +55,48 @@ export const useGetGroups = ({
...
+    const queryString = urlSearchParams.toString();
+    return useSnackQuery({
+        queryKey: ['groups', queryString],
+        queryFn: () => getRequest(`/api/groups/dropdown/?${queryString}`),
+        snackErrorMsg: MESSAGES.fetchGroupsError,
+        options: {
+            staleTime,
+            select: data => {
+                if (!data) return [];
+                return data.groups.map(group => {
```
Comment (hakifran) [inline hat/assets/js/apps/Iaso/domains/orgUnits/hooks/requests/useGetGroups.ts]: There is no groups in data, you should map directly data `TypeError: data.groups is undefined.`
Screenshot from 2023-05-15 10-04-54

**Review by hakifran — CHANGES_REQUESTED** [review]
And also there should be this `{
        isRequired: false,
        key: "orgUnitGroups"
    }` to be added in the /polio/js/config.js on campaignsFilters so that it can be taken in account while filtering on orgUnitGroups.

**Review by quang-le — CHANGES_REQUESTED** [review]
I get the response from the API, but I don't see anything in the dropdown list:
Screenshot 2023-05-15 at 18 27 40

**Review by quang-le — CHANGES_REQUESTED** [review]
In the embedded calendar, the dropdown shows, but the filter is not applied in the video, applying the filter should only keep the Benin campaign. It works in the normal calendar though)


https://github.com/BLSQ/iaso/assets/38907762/b4acc7b8-a5dd-4d87-9144-590bfa93502a

**Review by quang-le — APPROVED** [review]

**Pull request merged**